signer/hsm.go

```diff
@@ -107,6 +107,36 @@ type GCPHSM struct {
 	GenericHSM
 }
 
+type unlimitedBytesRandReader struct {
+	rr crypto11.PKCS11RandReader
+}
+
+func (ubrr *unlimitedBytesRandReader) Read(data []byte) (int, error) {
+	var n int
+	var err error
+	need := len(data)
+	for need > 0 {
+		var bytesRead int
+		next := make([]byte, min(need, 1024))
+		bytesRead, err = ubrr.rr.Read(next)
+		n += bytesRead
+		if err != nil {
+			return n, err
+		}
+		data = append(data, next...)
```

Comment:
I'm not sure how this is working correctly. It only appends to data, but does not overwrite existing data in `data`. That doesn't seem correct?

Reply:
I think going back to the copying version is fine, but we should figure out why the tests were able to pass.